mr/src/test/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandlerBaseTest.java

```diff
@@ -157,6 +177,282 @@ public void testJoinTables() throws IOException {
     Assert.assertArrayEquals(new Object[] {1L, "Bob", 102L, 33.33d}, rows.get(2));
   }
 
+  @Test
+  public void testCreateDropTable() throws TException, IOException {
+    // We need the location for HadoopTable based tests only
+    String location = locationForCreateTable(temp.getRoot().getPath(), "customers");
+    shell.executeStatement("CREATE EXTERNAL TABLE customers " +
+        "STORED BY 'org.apache.iceberg.mr.hive.HiveIcebergStorageHandler' " +
+        (location != null ? "LOCATION '" + location + "' " : "") +
+        "TBLPROPERTIES ('" + InputFormatConfig.TABLE_SCHEMA + "'='" + SchemaParser.toJson(CUSTOMER_SCHEMA) + "', " +
+        "'" + InputFormatConfig.PARTITION_SPEC + "'='" + PartitionSpecParser.toJson(IDENTITY_SPEC) + "', " +
+        "'dummy'='test')");
+
+    Properties properties = new Properties();
+    properties.put(Catalogs.NAME, TableIdentifier.of("default", "customers").toString());
+    if (location != null) {
+      properties.put(Catalogs.LOCATION, location);
+    }
+
+    // Check the Iceberg table data
+    org.apache.iceberg.Table icebergTable = Catalogs.loadTable(shell.getHiveConf(), properties);
+    Assert.assertEquals(SchemaParser.toJson(CUSTOMER_SCHEMA), SchemaParser.toJson(icebergTable.schema()));
+    Assert.assertEquals(PartitionSpecParser.toJson(IDENTITY_SPEC), PartitionSpecParser.toJson(icebergTable.spec()));
+    Assert.assertEquals(Collections.singletonMap("dummy", "test"), icebergTable.properties());
+
+    // Check the HMS table parameters
+    IMetaStoreClient client = null;
+    org.apache.hadoop.hive.metastore.api.Table hmsTable;
```

Comment:
We might want to use a shared `HiveClientPool` for this. We've had problems in the past where too many clients led to the HMS becoming unresponsive in tests. It's really annoying and makes tests flaky. Sharing a pool across all tests fixes the problem, and makes us more confident that if we hit a connection issue, it is probably in prod code and not test.

I think it would also make the test cases smaller because you wouldn't need try/finally blocks.

Reply:
We are starting a `new TestHiveMetastore()` in the `before` method. So every test uses its own HMS instance.
So I think reusing HMS connection is not really an option until we use the same HMS instance throughout the tests

Reply:
Oh, we aren't using a shared metastore across test cases?

We should probably fix that eventually (maybe get this PR in first though). We do want to keep the metastore around long enough that we detect Iceberg connection leaks. Although it is annoying to have tests fail because the metastore hangs, it's better than releasing a version that leaves connections open and takes down a metastore.

Reply:
It would also save a serious amount of testing time.
I think it might need changes in HiveRunner though, we have to check

Reply:
We just made the change to use a suite-level metastore in #1478 and it appears to work fine. We should explore doing the same here, but probably in a follow-up.

Reply:
Rebased to the master, and using the suite-level metastore